File: examples/example-42/java/LoggingStep.java
import io.cucumber.java.en.Given;
import org.apache.logging.log4j.Logger;

public class LoggingStep {
    private static final Logger log = LogManager.getLogger(MySteps.class);

    @Given("test values are logged")
    public void given_test_values_are_logged() {
        log.debug("this message will not be seen because of the log level.");
        log.info("this message will be in the file but not shown on the screen.");
        log.warn("this message will be in both the file and on the screen.");
    }
}

File: examples/example-42/js/logging_step.js
const { Given } = require('@cucumber/cucumber');
const { log } = require('../logging');

Given('test values are logged', function() {
    log.debug('this message will not be seen because of the log level.');
    log.info('this message will be in the file but not shown on the screen.');
    log.warn('this message will be in both the file and on the screen.');
});